package/gluon-core/luasrc/usr/lib/lua/gluon/wireless.lua

```diff
@@ -39,13 +39,26 @@ local function get_wlan_mac_from_driver(radio, vif)
 	return addresses[vif+1]
 end
 
-function M.get_wlan_mac(radio, index, vif)
-	local addr = get_wlan_mac_from_driver(radio, vif)
-	if addr then
-		return addr
+local radio_mac_offsets = {
+	client = 0,
+	mesh = 1,
+	owe = 2,
+	wan_radio = 3,
+}
+
+function M.get_wlan_mac(func, index, radio)
+	local offset = radio_mac_offsets[func]
+	if offset == nil then
+		return nil
+	end
+	if radio then
```

Comment:
Why is this `if radio` needed? It wasn't before

Reply:
The `private-wlan.rst` now uses `$(lua -e "print(require('gluon.wireless').get_wlan_mac('wan_radio', $RID))")`.

If we would not check for radio being nil, this would fail - or we would have to provide an empty table to get_wlan_mac.

I liked to have the user facing call simpler, so I went to make the radio table optional.

Reply:
> The `private-wlan.rst` now uses `$(lua -e "print(require('gluon.wireless').get_wlan_mac('wan_radio', $RID))")`.
> 
> If we would not check for radio being nil, this would fail - or we would have to provide an empty table to get_wlan_mac.
> 
> I liked to have the user facing call simpler, so I went to make the radio table optional. This is new, as we called generate_mac before at this place.

Mmmh, the old code is simply wrong though, and the new one needs a proper radio. If the WLAN adapter provides multiple MAC addresses, those *must* be used instead of the generated ones (I know at least ralink doesn't allow generated MAC addresses, maybe also (some) Mediatek)

Reply:
Hmm... we're duplicating more and more complex logic, but it would be better to just generate the whole section, using the same code for both the web UI and manually. This could be achieved by moving the actual settings to `/etc/config/gluon` and regenerate the whole of `/etc/config/wireless` on every reconfigure - but that is definitely out of scope for this PR. So maybe we should just keep the incorrect code for now...

Reply:
Thanks for your patience :)